R/designOfExperiments.R

```diff
@@ -158,36 +156,40 @@ designOfExperiments <- function(jaspResults, dataset, options, ...){
                                "factorialCornerReplicates",
                                "factorialBlocks",
                                "designBy",
-                               "factorialResolution"))
-
-    table$addColumnInfo(name = 'runOrder', title = gettext("RunOrder"), type = 'string')
+                               "factorialResolution",
+                               "numberOfFactors"))
 
     results <- options[["factors"]]
 
     factorNames <- factorLows <- factorHighs <- character()
     factorVectors <- list()
 
     for(i in 1:length(results)){
-      factorNames[i]      <- results[[i]]$name
+      factorNames[i]      <- results[[i]]$factorName
       factorLows[i]       <- results[[i]]$low
-      factorHighs[i]      <- results[[i]]$high
+      factorHighs[i]      <- results[[i]]$high1
       factorVectors[[i]]  <- c(factorLows[i], factorHighs[i])
-      table$addColumnInfo(name = factorNames[i], title = factorNames[i], type = 'string')
     }
 
-    table$addColumnInfo(name = 'measurement', title = gettext("Measurement"), type = 'string')
-
-    jaspResults[["displayDesign"]] <- table
-
-    # This is really bare-bones (only works for 2 factors)
-    runOrder <- 1:8
-    factor1  <- factor(c(rep(factorVectors[[1]][1], each = as.numeric(options[["factorialRuns"]])), rep(factorVectors[[1]][2], each = as.numeric(options[["factorialRuns"]]))), levels = factorVectors[[1]])
-    factor2  <- factor(c(rep(factorVectors[[2]][1], times = as.numeric(options[["factorialRuns"]])), rep(factorVectors[[2]][2], times = as.numeric(options[["factorialRuns"]]))), levels = factorVectors[[2]])
+    table$addColumnInfo(name = 'runOrder', title = gettext("RunOrder"), type = 'string')
+    runOrder <- 1:2^as.numeric(options[["numberOfFactors"]])
+
+    rows <- data.frame(runOrder = runOrder)
+    rows <- cbind.data.frame(rows,
+                             matrix(0,
+                                    ncol = as.numeric(options[["numberOfFactors"]]),
+                                    nrow = nrow(rows),
+                                    dimnames = NULL))
+
+    for(i in 1:as.numeric(options[["numberOfFactors"]])){
+      rows[,i+1] <- rep(c(rep(1, 2^(as.numeric(options[["numberOfFactors"]])-i)), rep(-1, 2^(as.numeric(options[["numberOfFactors"]])-i))), 2^(i-1))
+      colnames(rows)[i+1] <- paste("Factor", as.character(i), sep = " ")
+      #table$addColumnInfo(name = factorNames[i], title = factorNames[i], type = 'number') #uncomment when 'factors' (i.e., the row 2 issue) is fixed in the QML file
+    }
 
-    rows <- data.frame(runOrder = runOrder,
-                       factor1 = factor1,
-                       factor2 = factor2)
-    colnames(rows) <- c("runOrder", factorNames)
+    colnames(rows) <- as.character(colnames(rows)) #when 'the row 2 issue' is fixed, this line should instead be "colnames(rows) <- c("runOrder", factorNames)
     table$setData(rows)
+
+    jaspResults[["displayDesign"]] <- table
```

Comment:
It's better to provide the table to `jaspResults` right after you have created all the columns so that it can render some empty cells before you fill it. So you can move this up a little bit.